src/Microsoft.Health.SqlServer/Features/Client/SqlConnectionWrapper.cs

```diff
@@ -60,8 +60,7 @@ internal async Task InitializeAsync(string initialCatalog = null, CancellationTo
         }
         else
         {
-            SqlConnection = await _sqlConnectionBuilder.GetSqlConnectionAsync(initialCatalog, _sqlServerDataStoreConfiguration.MaxPoolSize, cancellationToken: cancellationToken).ConfigureAwait(false);
-            SqlConnection.RetryLogicProvider = _sqlRetryLogicBaseProvider;
```

Comment:
I see you have removed the retry logic provider and is it just sufficient to have `services.AddSqlRetryLogicProvider();` this in the registration? Did we test out if in case of timeout, we retry the specified number of times?